nrf52840dk_layout.ld

```diff
@@ -3,8 +3,10 @@
  */
 
 MEMORY {
-  /* The application region is 64 bytes (0x40) */
-  FLASH (rx) : ORIGIN = 0x00040040, LENGTH = 0x000BFFC0
+  /* The application region is 64 bytes (0x40) and we reserve 0x40000 at the end
+   * of the flash for the persistent storage.
+   */
```

Comment:
What about the layout for the other boards we support? Is this layout file used for all of them (in which case we should probably rename it `nrf52840_layout`, as a property of the chip rather than the board)? How are dfu-based boards affected?

cc @jmichelp

Reply:
Indeed the layout depends on the chip, not the board it's soldered on. And because we only support the nrf52840 chip so far, that's the only layout file we need (it's used to compile the app, not the kernel).

If we rename it, we need to also change `deploy.py` accordingly.

Reply:
I created #105 for that.